src/lib/ext-activitylog.js

```diff
@@ -15,8 +15,14 @@ class Model {
     this.logs.push(...logs);
   }
 
-  setFilter({ logKey, valueEquals }) {
-    this.filter[logKey] = valueEquals;
+  /**
+   * Sets the values of any key in filter object
+   * @param {string} logKey - The key in the filter object.
+   * @param {Set} filterDescriptor - It contains the active checkboxes.
```

Comment:
This description is too specific to the UI. The model doesn't need to know about how the filter is created. I intentionally chose a variable name (`filterDescriptor`) that clearly describes the role of the value, independent of the DOM.

The type definition is incorrect. For `logKey="keyword"`, `filterDescriptor` is expected to be a string. This mistake shows why the need for proper documentation is important: if even you, who wrote all code, can't get the type right at the first try, imagine the difficulty that others (who don't work on the code) will have at trying to figure out the parameters to pass.

The JSDoc here should be written such that anyone who isn't familiar with the whole code base is able to know what kind of values they are allowed to pass (and if not obvious - why). The main purpose is to help the reader with understanding how they can use the function.

Reply:
Thanks for explaining. I have updated the JSDoc description :-)